### PR TITLE
fix: correct TypeScript parsing issues

### PR DESCRIPTION
**Pull request by htunnicliff**
## Changes

- Added `@typescript-eslint/parser` as a dependency.
- Switched TypeScript file glob to match Next.js's pattern.
- Added documentation for troubleshooting multiple `tsconfig.json` files.

### CODE REVIEW

File: packages/eslint-config-typescript/package.json [hunk 1]
```diff
@@ -8,17 +8,20 @@
...
     "url": "http://github.com/reside-eng/lint-config/issues"
   },
   "dependencies": {
-    "@typescript-eslint/eslint-plugin": "5.18.0"
+    "@typescript-eslint/eslint-plugin": "5.19.0",
+    "@typescript-eslint/parser": "5.19.0",
+    "eslint-import-resolver-typescript": "2.7.1"
   },
   "peerDependencies": {
-    "@typescript-eslint/parser": ">=5.18.0",
```
Comment (prescottprue) [inline packages/eslint-config-typescript/package.json:20]: Do we not already have this installed everywhere we are using Typescript? Was there a specific issue that installing fixes other than matching the two?

File: packages/eslint-config-typescript/rules/import.js [hunk 2]
```diff
@@ -1,10 +1,16 @@
 module.exports = {
   settings: {
+    'import/parsers': {
+      [require.resolve('@typescript-eslint/parser')]: ['.ts', '.tsx', '.d.ts'],
```
Comment (prescottprue) [inline packages/eslint-config-typescript/rules/import.js:4]: 👏

File: packages/eslint-config-typescript/package.json (hunk 1 above)
Comment (htunnicliff) [reply packages/eslint-config-typescript/package.json:20]: Yes and no. I ran into resolution issues when testing the new config out with other repos. After trying different attempts, I noticed that Next.js's config chose to add `@typescript-eslint/parser` as a dependency.

**Comment by htunnicliff** [comment]
@prescottprue any further changes I should make?

**Review by prescottprue — APPROVED** [review]

**Pull request merged**